packages/vite/src/node/plugins/asset.ts

```diff
@@ -190,7 +191,8 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
         for (const file in bundle) {
           if (
             bundle[file].type === 'asset' &&
-            !file.includes('ssr-manifest.json')
+            !file.endsWith('ssr-manifest.json') &&
+            !jsSourceMapRE.test(file)
```

Comment:
Woops, this transformation is actually not correct. Which is why I had originally refactored things to split the asset check out. It needs to be (isAsset && (isSsrManifest || isJsSourceMap)) and that got hard to parse. Fixing now...